test/DotnetIsolatedE2EApps/MainApp/app.csproj

```diff
@@ -0,0 +1,32 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+  <PropertyGroup>
+    <TargetFramework>net8.0</TargetFramework>
+    <AzureFunctionsVersion>v4</AzureFunctionsVersion>
+    <OutputType>Exe</OutputType>
+    <ImplicitUsings>enable</ImplicitUsings>
+    <Nullable>enable</Nullable>
+  </PropertyGroup>
+  <ItemGroup>
+    <FrameworkReference Include="Microsoft.AspNetCore.App" />
+    <PackageReference Include="Microsoft.Azure.Functions.Worker" Version="1.21.0" />
+    <PackageReference Include="Microsoft.Azure.WebJobs.Extensions.DurableTask" Version="3.1.2" />
+    <PackageReference Include="Microsoft.Azure.Functions.Worker.Extensions.DurableTask" Version="1.3.2" />
```

Comment:
Is it possible to add a project reference to the Worker Extensions package like [this example](https://github.com/Azure/azure-functions-durable-extension/blob/53abf0f5d893340e2ebeb24e352377c5d0c15393/test/SmokeTests/OOProcSmokeTests/DotNetIsolated/DotNetIsolated.csproj#L24) instead of referencing a specific version?

Reply:
Agreed, this would solve several issues with the current solution, including: 
- Dynamic modifying of app.csproj with the build script will lead to unnecessary noise in commits and/or work excluding these files from unrelated PRs
- Would remove the requirement of manually bumping extension versions to run test locally
- Would simplify the build script 

Need to explore this further. The extension build process in isolated has some gotchas but it may be possible to work around

Reply:
It would definitely be better to do a project reference as I mentioned in a separate comment, but we can do that as part of a follow up PR if it's not a simple fix.

Reply:
Done!